core/renderers/common/path_object.ts

```diff
@@ -273,7 +272,21 @@ export class PathObject implements IPathObject {
     offset: Coordinate,
     rtl: boolean,
   ) {
-    if (this.connectionHighlights.has(connection)) return;
+    if (this.connectionHighlights.has(connection)) {
+      const currPath = this.connectionHighlights
+        .get(connection)
+        ?.getAttribute('d');
+      const currOffset = this.highlightOffsets.get(connection);
+      if (
+        connectionPath === currPath &&
+        Coordinate.equals(offset, currOffset)
+      ) {
+        return;
+      } else {
```

Comment:
Can remove this `else`. Also could make this a little clearer/more self documenting by extracting the `if` clause into a `highlightsMatch` function so

```
if (this.connectionHighlights.has(connection)) {
  if (this.highlightsMatch(connectionPath, offset, existingHighlight) return;
  this.removeConnectionHighlight(connection);
}

// ... rest of addConnectionHighlight ...


highlightsMatch(connectionPath, offset, existingHighlight) {
  const existingPath = existingHighlight.get(connection)?.getAttribute('d');
  const existingOffset = this.highlightOffsets.get(connection);
  return connectionPath === existingPath && Coordinate.equals(offset, existingOffset);
}
```

just a suggestion though :) feel free to disagree

Reply:
Much better! Done.